front/lib/dust_api.ts

```diff
@@ -0,0 +1,69 @@
+import { Project } from "@app/types/project";
+
+const { DUST_API: DUST_API_URL } = process.env;
+
+type ErrorRsponse = { error: string };
+type SuccessResponse<T> = { response: T };
+
+type DustAPIResponse<T> = ErrorRsponse | SuccessResponse<T>;
+
+export function isErrorResponse<T>(
+  response: DustAPIResponse<T>
+): response is { error: string } {
+  return "error" in response;
+}
+
+type GetDatasetsResponse = {
+  datasets: { [key: string]: { hash: string; created: number }[] };
+};
+
+type GetDatasetResponse = {
+  dataset: {
+    created: number;
+    dataset_id: string;
+    hash: string;
+    keys: string[];
+    data: { [key: string]: any }[];
+  };
+};
+
+export const DustAPI = {
+  async createProject(): Promise<DustAPIResponse<{ project: Project }>> {
+    const response = await fetch(`${DUST_API_URL}/projects`, {
+      method: "POST",
+    });
+    return response.json() as Promise<DustAPIResponse<{ project: Project }>>;
+  },
+
+  async getDatasets(
+    projectId: string
+  ): Promise<DustAPIResponse<GetDatasetsResponse>> {
```

Comment:
We should use @lasryaric 's Error/Ok framework here?

Reply:
@lasryaric @spolu  can you check https://github.com/dust-tt/dust/pull/229/commits/643a36879ff1ab06019d669f22cbc275c73cbe7a ? It updates the `Result` framework to use type guards (the compiler forces you at compile time to check if there's an error before you can use `value`, slightly safer than having an exception at runtime)

Reply:
Yes done in the commit right after